plugins/modules/ndo_l3out_interface_group_policy.py

```diff
@@ -0,0 +1,512 @@
+
+DOCUMENTATION = r"""
+---
+module: ndo_l3out_interface_group_policy
+short_description: Manage L3Out Interface Group Policy on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage L3Out Interface Group Policy on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Shreyas Srish (@shrsr)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be an L3Out template.
+    type: str
+    aliases: [ l3out_template ]
+    required: true
+  l3out:
+    description:
+    - The name of the L3Out.
+    type: str
+    aliases: [ l3out_name ]
+    required: true
+  description:
+    description:
+    - The description of the L3Out Interface Group Policy.
+    type: str
+  name:
+    description:
+    - The name of the L3Out Interface Group Policy.
+    type: str
+    aliases: [ l3out_interface_group_policy ]
+  interface_routing_policy:
+    description:
+    - The configuration of the Interface Routing Policy.
+    - Providing an empty dictionary will remove the O(interface_routing_policy={}) from L3Out Interface Group Policy.
+    type: dict
+    suboptions:
+      name:
+        description:
+        - The name of the L3Out Interface Routing Policy.
+        type: str
+      template:
+        description:
+        - The template associated with the  L3Out Interface Routing Policy.
+        type: str
+  interface_routing_policy_uuid:
+    description:
+    - The UUID of the L3Out Interface Routing Policy.
+    - Providing an empty string will remove the O(interface_routing_policy_uuid="") from L3Out Interface Group Policy.
+    type: str
+  bfd:
+    description:
+    - The BFD configuration.
+    - Providing an empty dictionary will remove the O(bfd={}) from the L3Out Interface Group Policy.
+    type: dict
+    suboptions:
+      enable:
+        description:
+        - Whether BFD is enabled.
+        type: bool
+      authentication:
+        description:
+        - Whether BFD authentication is enabled.
+        type: bool
+      key_id:
+        description:
+        - The BFD key ID.
+        type: int
+      key:
+        description:
+        - The BFD key.
+        type: str
+  bfd_multi_hop:
+    description:
+    - The BFD multi-hop configuration.
+    - Providing an empty dictionary will remove the O(bfd_multi_hop={}) from the L3Out Interface Group Policy.
+    type: dict
+    suboptions:
+      enable:
+        description:
+        - Whether BFD multi-hop is enabled.
+        type: bool
+      authentication:
+        description:
+        - Whether BFD multi-hop authentication is enabled.
+        type: bool
+      key_id:
+        description:
+        - The BFD multi-hop key ID.
+        type: int
+      key:
+        description:
+        - The BFD multi-hop key.
+        type: str
+  ospf:
+    description:
+    - The OSPF configuration.
```

Comment:
should we elaborate OSPF just at line 111?

Reply:
I think we have sub options here which explain more..